usr/iscsi/iscsi_tcp.c

```diff
@@ -494,26 +533,46 @@ static int iscsi_tcp_conn_login_complete(struct iscsi_connection *conn)
 static size_t iscsi_tcp_read(struct iscsi_connection *conn, void *buf,
 			     size_t nbytes)
 {
+#ifndef USE_NET_IN_STREAM
 	struct iscsi_tcp_connection *tcp_conn = TCP_CONN(conn);
 	return read(tcp_conn->fd, buf, nbytes);
+#else
+	ssize_t rc;
+	rc = net_is_read(conn->in_stream, buf, nbytes);
+	if (rc) {
+		return rc;
+	}
+	if (net_is_closed(conn->in_stream)) {
+		errno = ECONNRESET;
+		return 0;
+	}
+	errno = net_is_last_error(conn->in_stream);
+	return -1;
+#endif
 }
 
 static size_t iscsi_tcp_write_begin(struct iscsi_connection *conn, void *buf,
 				    size_t nbytes)
 {
+#ifndef USE_NET_OUT_STREAM
 	struct iscsi_tcp_connection *tcp_conn = TCP_CONN(conn);
-	int opt = 1;
-
-	setsockopt(tcp_conn->fd, SOL_TCP, TCP_CORK, &opt, sizeof(opt));
 	return write(tcp_conn->fd, buf, nbytes);
+#else
+	ssize_t rc = net_os_write(conn->out_stream, buf, nbytes);
+	if (rc) {
+		return rc;
+	}
+	if (net_os_closed(conn->out_stream)) {
+		errno = ECONNRESET;
+		return -1;
+	}
+	errno = net_os_last_error(conn->out_stream);
+	return -1;
+#endif
 }
 
 static void iscsi_tcp_write_end(struct iscsi_connection *conn)
```

Comment:
Is this function never gets called?

Reply:
we cannot remove this function. Look for calls to ep_write_end().